src/lib/utils/project/GitProjectService.ts

```diff
@@ -327,97 +329,102 @@ export default class GitProjectService {
-        const receiver = {
-          type: 'dripList' as const,
-          weight,
-          accountId: itemId,
-        };
+      const scaledWeight = Math.floor(
+        Math.floor(weight * (highLevelPercentages['dependencies'] / 100)),
+      );
 
-        dependenciesSplitMetadata.push(receiver);
-        receivers.push(receiver);
+      switch (item.type) {
+        case 'address': {
+          const receiver = {
+            sublist: 'dependencies' as const,
+            type: 'address' as const,
+            weight: scaledWeight,
+            accountId: accountId,
+          };
+
+          receivers.push(receiver);
+          break;
+        }
+        case 'drip-list': {
+          const receiver = {
+            sublist: 'dependencies' as const,
+            type: 'dripList' as const,
+            weight: scaledWeight,
+            accountId: accountId,
+          };
+
+          receivers.push(receiver);
+          break;
+        }
+        case 'project': {
+          const receiver = {
+            sublist: 'dependencies' as const,
+            type: 'repoDriver' as const,
+            weight: scaledWeight,
+            accountId: accountId,
+            source: GitProjectService.populateSource(
+              item.project.source.forge,
+              item.project.source.repoName,
+              item.project.source.ownerName,
+            ),
+          };
+
+          receivers.push(receiver);
+          break;
+        }
       }
     }
 
-    // Populate maintainers splits and metadata.
-    const maintainersInput = Object.entries(maintainerListEditorConfig.percentages);
-
-    const maintainersSplitsMetadata: LatestVersion<
-      typeof repoDriverAccountMetadataParser
-    >['splits']['maintainers'] = [];
+    for (const [accountId, weight] of Object.entries(maintainerListEditorConfig.weights)) {
+      if (weight === 0) continue;
 
-    for (const [address, percentage] of maintainersInput) {
-      const weight = Math.floor(
-        (Number(percentage) / 100) * 1000000 * (highLevelPercentages['maintainers'] / 100),
+      const scaledWeight = Math.floor(
+        Math.floor(weight * (highLevelPercentages['maintainers'] / 100)),
       );
 
-      if (weight === 0) continue;
-
       const receiver = {
+        sublist: 'maintainers' as const,
         type: 'address' as const,
-        weight,
-        accountId: await this._addressDriverClient.getAccountIdByAddress(address),
+        weight: scaledWeight,
+        accountId,
       };
 
-      maintainersSplitsMetadata.push(receiver);
       receivers.push(receiver);
     }
 
+    // Adjust weights to ensure no tiny remainder
+    const MAX_WEIGHT = 1000000;
+
+    function adjustWeights<T extends { weight: number }>(input: T[]): T[] {
+      const totalWeight = Object.values(input).reduce((acc, { weight }) => acc + weight, 0);
+      const remainder = MAX_WEIGHT - totalWeight;
+
+      if (remainder > 0 && remainder < 5) {
+        input[0].weight += remainder;
```

Comment:
Wouldn't this throw if the input is empty?

Reply:
I think it's OK to assume it's not empty here bc a project can't have no receivers.